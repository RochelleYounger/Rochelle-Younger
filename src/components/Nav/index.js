import React from 'react';

function Nav() {

  const handleClick = (item) => {
    console.log(item);
    return item;
  };
  return (
    <header>
      <h2>
        <a data-testid='link' href='/'>Rochelle Younger</a>
      </h2>
      <nav>
        <ul className='flex-row'>
          <li className='mx-2'>
            <a data-testid='about' href='#about' onClick={() => handleClick('About')}>
              About me
            </a>
          </li>
          <li className='mx-2'>
            <span onClick={() => handleClick('Contact')}>Contact</span>
          </li>
          <li className='mx-2'>
            <span
            key='projects'
            onClick={() => handleClick('Projects')} >Projects</span>
          </li>
        </ul>
      </nav>
    </header>
  );
}

export default Nav;