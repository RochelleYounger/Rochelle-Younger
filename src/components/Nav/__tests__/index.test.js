import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';
afterEach(cleanup);

describe('Nav component', () => {
  it('Baseline test', () => {
    render(<Nav />);
  });

  // Second Test
  it('DOM node structure matches snapshot', () => {
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('links are visible', () => {
    it('inserts text into the links', () => {
      const { getByTestId } = render(<Nav />);
      expect(getByTestId('link')).toHaveTextContent('Rochelle Younger');
      expect(getByTestId('about')).toHaveTextContent('About me');
    });
  })
})