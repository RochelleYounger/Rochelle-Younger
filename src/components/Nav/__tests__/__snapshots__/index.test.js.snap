// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component DOM node structure matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        Rochelle Younger
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li
          class="mx-2"
        >
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-2"
        >
          <span>
            Projects
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
